hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneClientAdapterImpl.java

```diff
@@ -549,6 +558,31 @@ private FileStatusAdapter toFileStatusAdapter(OzoneFileStatus status,
     );
   }
 
+  private FileStatusAdapter toFileStatusAdapter(OzoneFileStatusLight status,
+                                                String owner, URI defaultUri, Path workingDir) {
+    BasicOmKeyInfo keyInfo = status.getKeyInfo();
+    short replication = (short) keyInfo.getReplicationConfig()
+        .getRequiredNodes();
+    return new FileStatusAdapter(
+        keyInfo.getDataSize(),
+        keyInfo.getReplicatedSize(),
+        new Path(OZONE_URI_DELIMITER + keyInfo.getKeyName())
+            .makeQualified(defaultUri, workingDir),
+        status.isDirectory(),
+        replication,
+        status.getBlockSize(),
+        keyInfo.getModificationTime(),
+        keyInfo.getModificationTime(),
+        status.isDirectory() ? (short) 00777 : (short) 00666,
```

Comment:
Existing `toFileStatusAdapter(OzoneFileStatus, ...)` is duplicated here almost exactly, so I think we should make improvements like this in a follow-up task:

- create constants
- reduce duplication if possible (two methods within same class, as well as other FS adapter implementation)

Reply:
I agree, this can be addressed as a follow-up task!

Reply:
yeah we should pursue this in a followup task for this. There is too much duplication b/w BasicOzoneFIleSystem & BasicRootedOzoneFileSystem, BasicOzoneClientAdapterImpl & BasicRootedOzoneClientAdapterImpl. This created a lot of confusion while implementing it.

Reply:
I also don't like the way BasicKeyInfo class has been definied. Ideally we should have OzoneFileStatus extend OzoneFileStatusLight. It would make our code much more modular and avoid such unnecessary duplication of code.